.github/workflows/ci.yaml

```diff
@@ -0,0 +1,64 @@
+name: Continuous Integration
+on: [push, pull_request]
+
+jobs:
+  lint:
+    runs-on: ubuntu-latest
+    steps:
+      - uses: actions/checkout@v3
+      - uses: actions/setup-go@v3
+        with:
+          go-version: 1.19
+      - name: Run linters
+        uses: golangci/golangci-lint-action@v3.4.0
+        with:
+          version: v1.50.1 # The next version requires Go 1.20
+
+  integration:
+    runs-on: ubuntu-latest
+    strategy:
+      matrix:
+        go: ["1.18.x", "1.19.x", "1.20.x"]
```

Comment:
It looks like this matrix is not actually being used because you set `go-version: '1.20'` later on? (I'm sorry if I am missing something)

https://github.com/actions/setup-go#matrix-testing